File: flutter/storyly_placement_flutter/ios/Classes/Common/Data/Analytics/AnalyticsData.swift
import Foundation
import StorylyAnalytics

func decodeSTRAnalyticsConfig(_ dict: [String: Any]) -> STRAnalyticsConfig? {
    guard let token = dict["token"] as? String else { return nil }

    let builder = STRAnalyticsConfig.Builder()
    if let userId = dict["userId"] as? String {
        _ = builder.setUserId(userId: userId)
    }
    return builder.build(token: token)
}

func decodeSTRAnalyticProductEvent(_ event: String?) -> STRAnalyticProductEvent? {
    guard let event = event else { return nil }
    return STRAnalyticProductEvent(value: event)
}

func decodeSTRAnalyticProduct(_ dict: [String: Any]) -> STRAnalyticProduct? {
    guard let productId = dict["productId"] as? String,
          let productGroupId = dict["productGroupId"] as? String,
          let title = dict["title"] as? String,
          let price = dict["price"] as? Double else {
        return nil
    }

    let desc = dict["desc"] as? String
    let salesPrice = dict["salesPrice"] as? Double
    let quantity = dict["quantity"] as? Int ?? 1

    return STRAnalyticProduct(
        productId: productId,
        productGroupId: productGroupId,
        title: title,
        desc: desc,
        price: Float(price),
        salesPrice: salesPrice as NSNumber?,
        quantity: quantity
    )
}

File: flutter/storyly_placement_flutter/android/src/main/kotlin/com/appsamurai/storyly/storyly_placement_flutter/common/data/analytics/AnalyticsData.kt
package com.appsamurai.storyly.storyly_placement_flutter.common.data.analytics

import com.appsamurai.storyly.analytics.config.STRAnalyticsConfig
import com.appsamurai.storyly.analytics.model.STRAnalyticProduct
import com.appsamurai.storyly.analytics.model.STRAnalyticProductEvent

internal fun decodeSTRAnalyticsConfig(config: Map<String, Any?>): STRAnalyticsConfig? {
    val token = config["token"] as? String ?: return null
    val builder = STRAnalyticsConfig.Builder()
    (config["userId"] as? String)?.let { builder.setUserId(it) }
    return builder.build(token)
}

internal fun decodeSTRAnalyticProductEvent(event: String?): STRAnalyticProductEvent? {
    event ?: return null
    return STRAnalyticProductEvent.entries.firstOrNull { it.name == event }
}

internal fun decodeSTRAnalyticProduct(product: Map<String, Any?>?): STRAnalyticProduct? {
    product ?: return null
    val productId = product["productId"] as? String ?: return null
    val productGroupId = product["productGroupId"] as? String ?: return null
    val title = product["title"] as? String ?: return null
    val price = (product["price"] as? Number)?.toFloat() ?: return null
    return STRAnalyticProduct(
        productId = productId,
        productGroupId = productGroupId,
        title = title,
        desc = product["desc"] as? String,
        price = price,
        salesPrice = (product["salesPrice"] as? Number)?.toFloat(),
        quantity = (product["quantity"] as? Number)?.toInt() ?: 1,
    )
}
